### PR TITLE
Add cache headers to static files served through Rails

### LINKED ISSUE #24021: Some static assets do not have proper cache-control headers when served by Rails
### Steps to reproduce the problem

When `RAILS_SERVE_STATIC_FILES` is set, Rails is not setting any `Cache-Control` header for some files.

For example:
- `/oops.png`
- `/robots.txt`
- `/favicon.ico`
- `/sounds/boop.mp3`
- `/ocr/lang-data/eng.traineddata.gz`
- `/emoji/*`

Packs and other assets have a correct `Cache-Control` header set. This is detrimental to caching when using a CDN or reverse proxy cache, and should decrease the load on origins quite a lot.

### Expected behaviour

Static files have a correct `Cache-Control` header

### Actual behaviour

They dont

### Detailed description

This can be fixed by custom configuration on your proxies, but I think it should be handled out of the box.

### Specifications

Mastodon 4.1.0, using the official Docker container.

### PR DESCRIPTION
**Pull request by Gargron**
Fix #24021

### CODE REVIEW

**Review by ykzts — APPROVED** [review]

File: config/environments/production.rb [hunk 1]
```diff
@@ -67,7 +57,7 @@
 
   # Enable locale fallbacks for I18n (makes lookups for any locale fall back to
   # English when a translation cannot be found).
-  config.i18n.fallbacks = [:en]
+  config.i18n.fallbacks = true
```
Comment (ClearlyClaire) [inline config/environments/production.rb:60]: Why this change?
